src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +112,872 @@ static DamageTable D_80AD2550 = {
+        case 0x1FDB:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD144C.s")
+        case 0x1FDD:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD14C8.s")
+        case 0x1FDF:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD1634.s")
+        case 0x1FE1:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD16A8.s")
+        case 0x1FE7:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD19A0.s")
+        case 0x1FF1:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD1A4C.s")
+        case 0x1FF3:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/EnOsn_Init.s")
+        case 0x1FF7:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/EnOsn_Destroy.s")
+        case 0x1FF9:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/EnOsn_Update.s")
+        case 0x1FFB:
+            this->unk_1F4 = 0x1FD3U;
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD1DA8.s")
+        case 0x1FD9:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/func_80AD1E28.s")
+        case 0x1FE3:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osn/EnOsn_Draw.s")
+        case 0x1FE9:
+
+        case 0x1FEB:
+
+        case 0x1FED:
+
+        case 0x1FEF:
+
+        case 0x1FF5:
+
+        case 0x1FFD:
+            this->unk_1F4 = 0x1FD4U;
+            break;
+
+        case 0x1FE5:
+            this->unk_1F4 = 0x1FFEU;
+            break;
+
+        case 0x1FD3:
+
+        case 0x1FD4:
+
+        case 0x1FFE:
+            this->unk_1F4 = 0x1FD5U;
+            this->unk_1EA |= 0x20;
+    }
+
+    func_801518B0(globalCtx, this->unk_1F4, &this->actor);
+}
+
+void func_80AD1398(EnOsn* this) {
+    this->cutscene = this->actor.cutscene;
+    if (((gSaveContext.inventory.items[SLOT_OCARINA] == ITEM_NONE) ||
+         (gSaveContext.inventory.items[gItemSlots[0x32]] == 0x32)) &&
+        ((this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene),
+          (gSaveContext.inventory.items[SLOT_OCARINA] != ITEM_NONE)) ||
+         (gSaveContext.inventory.items[gItemSlots[0x32]] == 0x32))) {
+        this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene);
+    }
+}
+
+void func_80AD144C(EnOsn* this, GlobalContext* globalCtx) {
+    u32 sp1C;
+
+    sp1C = Flags_GetSwitch(globalCtx, 0);
+    this->cutscene = this->actor.cutscene;
+    Actor_ChangeAnimation(&this->anime, sAnimations, 0);
+    if (sp1C == 0) {
+        this->actionFunc = func_80AD16A8;
+    } else {
+        this->actionFunc = func_80AD14C8;
+    }
+}
+
+void func_80AD14C8(EnOsn* this, GlobalContext* globalCtx) {
+    s16 temp_v1;
+    temp_v1 = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+    if ((gSaveContext.inventory.items[SLOT_OCARINA] != ITEM_NONE) &&
+        (((*(gBitFlags + 0xD)) & gSaveContext.inventory.questItems) == 0)) {
```

Comment:
`*(gBitFlags + 0xD)` is `gBitFlags[0xD]`. We really try to avoid that notation.
Taking a quick look at macros.h, you can see there's already a pattern for this, specifically
```c

#define CHECK_QUEST_ITEM(item) (((void)0, gSaveContext.inventory.questItems) & gBitFlags[item])
```
So, you can change this line to 
```suggestion
        (!CHECK_QUEST_ITEM(QUEST_SONG_HEALING))) {
```